docs/dxp/latest/en/developing-applications/core-frameworks/configurable-application/hiding-the-configuration-ui/resources/liferay-g8v3.zip/g8v3-impl/src/main/java/com/acme/g8v3/internal/configuration/admin/display/G8V3BakerConfigurationVisibilityController.java

```
@@ -0,0 +1,46 @@
+package com.acme.g8v3.internal.configuration.admin.display;
+
+import com.acme.g8v3.internal.configuration.G8V3AbleConfiguration;
+
+import com.liferay.configuration.admin.display.ConfigurationVisibilityController;
+import com.liferay.portal.configuration.metatype.annotations.ExtendedObjectClassDefinition;
+import com.liferay.portal.kernel.module.configuration.ConfigurationException;
+import com.liferay.portal.kernel.module.configuration.ConfigurationProvider;
+
+import java.io.Serializable;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Component(
+	property = "configuration.pid=com.acme.g8v3.internal.configuration.G8V3BakerConfiguration",
+	service = ConfigurationVisibilityController.class
+)
+public class G8V3BakerConfigurationVisibilityController
+	implements ConfigurationVisibilityController {
+
+	public boolean isG8V3BakerConfigurationEnabled() {
+		try {
+			_g8v3AbleConfiguration =
+				_configurationProvider.getSystemConfiguration(
+					G8V3AbleConfiguration.class);
+		}
+		catch (ConfigurationException configurationException) {
```

Should I be logging an exception here?

---

@mtokudome that's correct. As a general rule, we should always avoid "swallowing" an exception. Either it should be logged, handled with alternate logic, or rethrown.